test/if-then-else.test.js

```diff
@@ -368,3 +369,52 @@ t.test('nested if/then', t => {
     JSON.stringify({ a: 'A', bar: 'bar', bar1: 'bar1' })
   )
 })
+
+t.test('if/else with string format', (t) => {
+  t.plan(2)
+
+  const schema = {
+    if: { type: 'string' },
+    then: { type: 'string', format: 'date' },
+    else: { const: 'Invalid' }
+  }
+
+  const stringify = build(schema)
+
+  const date = new Date()
+
+  t.equal(stringify(date), `"${DateTime.fromJSDate(date).toISODate()}"`)
+  t.equal(stringify('Invalid'), '"Invalid"')
+})
+
+t.test('if/else with const integers', (t) => {
+  t.plan(2)
+
+  const schema = {
+    type: 'integer',
+    if: { minimum: 42 },
+    then: { const: 66 },
+    else: { const: 33 }
+  }
+
+  const stringify = build(schema)
+
+  t.equal(stringify(66.32), '66')
+  t.equal(stringify(33.12), '33')
```

Comment:
just to be sure that the rounding is not triggered

```suggestion
  t.equal(stringify(100.32), '66')
  t.equal(stringify(10.12), '33')
```